src/utils/common.ts

```diff
@@ -265,8 +265,12 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
   const voidSigner = new VoidSigner(senderAddress, provider);
 
   // Estimate the Gas units required to submit this transaction.
-  const nativeGasCost = gasUnits ? BigNumber.from(gasUnits) : await voidSigner.estimateGas(unsignedTx);
-  assert(nativeGasCost.gt(0), "Gas cost should not be 0");
+  const queries = [
+    gasUnits ? Promise.resolve(BigNumber.from(gasUnits)) : voidSigner.estimateGas(unsignedTx),
+    _gasPrice ? Promise.resolve({ maxFeePerGas: _gasPrice }) : getGasPriceEstimate(provider, chainId, transport)
+  ] as const;
+  let [nativeGasCost, { maxFeePerGas: gasPrice }] = await Promise.all(queries);
```

Comment:
This feels a bit like a bodge, but it seems to work OK. I'm open to another formulation if we can find something simpler, but I also anticipate that this will be removed entirely before too long (pending the gasPriceOracle being able to estimate gas internally - coming real soon now).